### DISCUSSION TITLE
Related Field class for representing external keys?

### DISCUSSION THREAD

**Discussion by rlittlefield** [django-json-api/django-rest-framework-json-api#999] [Q&A]
Has anyone had luck adapting one of the related field classes to take a single "something_id" field that is not defined as a foreign key in the database or model, and generate output that fits into the relationships object?

For example, if I have a Books model that has an author_id field, in which the Author model and related table don't exist. I still want to have a relationship property added for `"author": {"data": {"type": "users", "id": "1324"}}`.

I didn't see an option for this in the docs, as they all seem to require an actual model. If that isn't available, my next option is to see if there is way to generate a fake model in django for objects that don't actually live in the db. If those don't work, I'll see about making a new `ExternalKeyRelatedField` that can do this.

Update:
Subclassing `RelatedResourceField` not working so far. The superclass wants a model, as does the code in `utils.py` in `get_related_resource_type`. I don't think it is worth it to subclass the whole renderer.

Any suggestions?

**Comment by sliverc** [comment]
This question seems to be related to #942 where we are discussing what is currently working and what not.

**Reply by rlittlefield** [reply]
This might be a bit different because this is just a TextField not a GenericForeignKey. I already know the type, but the type doesn't actually have a table or model object anywhere.

**Reply by sliverc** [reply]
OK I see your point. This is certainly not gonna be easy and most likely you will need to have a custom json renderer class overwriting `extract_relationships`.
